### PR TITLE
Fps

### PR DESCRIPTION
**Pull request by setrofim**
Add rendering frames collection instruments

Add instruments to collect rendered frames data from Android devices from
surfaceflinger and gfxinfo.

### CODE REVIEW

**Comment by setrofim** [comment]
@setrofim test

**Comment by Sticklyman1936** [comment]
@setrofim Which Android versions will these work for?

**Comment by setrofim** [comment]
@Sticklyman1936 Should work with all of them — you should use the surfaceflinger version for older Androids, and the gfxinfo for the newer ones (M onwards, IIRC)

File: devlib/utils/rendering.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            writer.writerows(frames)
+
+    def collect_frames(self):
+        raise NotImplementedError()
+
+    def clear(self):
+        raise NotImplementedError()
+
+    def _process_raw_file(self):
+        raise NotImplementedError()
```
Comment (marcbonnici) [inline devlib/utils/rendering.py]: `_process_raw_file` and `collect_frames` have different arguments to their overridden counter parts.

File: devlib/exception.py [hunk 2]
```diff
@@ -49,3 +48,42 @@ def __init__(self, command, output):
...
+
+def get_traceback(exc=None):
+    """
+    Returns the string with the traceback for the specifiec exc
+    object, or for the current exception exc is not specified.
+
+    """
+    import StringIO, traceback
+    if exc is None:
+        exc = sys.exc_info()
```
Comment (marcbonnici) [inline devlib/exception.py]: Missing `sys` import.

**Comment by setrofim** [comment]
Issues addressed.

**Pull request merged**